src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -59,72 +80,131 @@ class AudioPlayerSub extends Component {
   }
 
   checkTrack = async () => {
-    // Check if the track has changed
+    /*
+    Check if the track in TrackPlayer (oldTrack) is different
+    than the track in props. This handles the case where a user
+    switches to a screen where the component is already rendered
+    (example: using the back button)
+    */
+    const {
+      topScreen,
+      track,
+      playing,
+      updatePlaying,
+      prevProgress,
+      duration,
+      autoplay,
+      automaticPause,
+    } = this.props
+
     const id = await TrackPlayer.getCurrentTrack()
     const oldTrack = await TrackPlayer.getTrack(id)
-    if (oldTrack?.url != this.props.track.url) {
+
+    if (topScreen && !this.state.switching && oldTrack?.url !== track.url) {
+      this.setState({ switching: true })
+
       const id = uuid()
-      const { track, playing, autoplay, updatePlaying } = this.props
       await TrackPlayer.add({
         id,
         url: track.url,
         title: track.title,
         artist: track.subtitle,
         artwork: track.artwork,
       })
-      TrackPlayer.skip(id)
+
+      await TrackPlayer.skip(id)
+
+      if (!automaticPause) {
```

Comment:
So if automaticPause is false, the TrackPlayer will automatically pause? I'm a little confused about this.

Reply:
Added a comment to it!

`// prevents previous screen's audio playing on new screens' audio player`

Reply:
Wouldn't you want it to pause when automatic pause is true then?